movies/movie_api_utils.py

```diff
@@ -85,3 +85,23 @@ def _get(self, url, **kwargs) -> dict[str, Any]:
             raise MovieApiException("Request to the movie database timed out.")
         except requests.exceptions.RequestException:
             raise MovieApiException("An error occurred while processing your request.")
+
+    def convert_date_string_into_object(self, movies, *, filter: str = None):
+        """
+        Iterates through a collection of movies and converts their release date string into a datetime object.
+        """
+        try:
+            movies_to_process = movies if filter is None else movies[filter]
```

Comment:
here is also a potential issue (if searching for a string pattern). You get a single result vs every result that matches your filter.

Reply:
Okay, looks like this is resolved.